### PR TITLE
Lower CMake minimum required version from 3.13.0 to 3.2.0

### PR DESCRIPTION
**Pull request by yourslab**
With these small changes, we can support CMake 3.2.0 or higher. Unfortunately, 3.1.0 has a bug in finding OpenSSL libraries, but CMake 3.2.0 has been tested to work. This change is especially useful for people with older Linux distros.

By submitting this pull request, I confirm that my contribution is made under the terms of the MIT license.

### CODE REVIEW

File: demos/shadow/shadow_demo_main/CMakeLists.txt [hunk 1]
```diff
@@ -12,14 +12,8 @@ include( ${CMAKE_SOURCE_DIR}/libraries/aws/device-shadow-for-aws-iot-embedded-sd
...
 
 # Demo target.
-add_executable(${DEMO_NAME})
-
-# Add to default target if all required macros needed to run this demo are defined
-check_aws_credentials(${DEMO_NAME})
-
-target_sources(
+add_executable(
     ${DEMO_NAME}
```
Comment (sarenameas) [inline demos/shadow/shadow_demo_main/CMakeLists.txt:16]: Do you want to update the indentation now that "PRIVATE" is gone?

Comment (yourslab) [reply demos/shadow/shadow_demo_main/CMakeLists.txt:16]: I do like having it indented though coz it kinda shows the heirarchy that these sources are under the executable.

File: platform/posix/transport/CMakeLists.txt [hunk 2]
```diff
@@ -52,8 +52,7 @@ target_link_libraries( openssl_posix
                        PUBLIC
                           sockets_posix
                        PRIVATE
-                          OpenSSL::Crypto
-                          OpenSSL::SSL
+                          ${OPENSSL_LIBRARIES}
```
Comment (aggarw13) [inline platform/posix/transport/CMakeLists.txt:55]: Where is `OPENSSL_LIBRARIES` variable defined?

Comment (yourslab) [reply platform/posix/transport/CMakeLists.txt:55]: It is defined by the built-in `FindOpenSSL.cmake` file once OpenSSL is found:
https://cmake.org/cmake/help/v3.6/module/FindOpenSSL.html

Comment (aggarw13) [reply platform/posix/transport/CMakeLists.txt:55]: Can you add a comment to provide a link to this page?

**Comment by aggarw13** [comment]
Should there be a mechanism to test that our CMake files are CMake 3.2.0 by updating the ci.yml file to use the lower version of Cmake?

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -17,7 +17,9 @@ jobs:
           submodules: recursive
       - name: CMake
```
Comment (aggarw13) [inline .github/workflows/ci.yml:18]:
```suggestion
      - name: Install CMake 3.2.0
```

Comment (aggarw13) [inline .github/workflows/ci.yml:18]:
```suggestion
      - name: Downloadn CMake 3.2.0
```

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -17,7 +17,9 @@ jobs:
...
         run: |
-          cmake -S . -B build/ \
+          curl https://cmake.org/files/v3.2/cmake-3.2.0-Linux-x86_64.tar.gz -o cmake.tar.gz
+          tar -xf cmake.tar.gz
+          cmake-3.2.0-Linux-x86_64/bin/cmake -S . -B build/ \
```
Comment (aggarw13) [inline .github/workflows/ci.yml:22]:
```suggestion
       - name: Build unit tests
          run: |   
          cmake-3.2.0-Linux-x86_64/bin/cmake -S . -B build/ \
```

**Review by aggarw13 — APPROVED** [review]

**Review by cobusve — APPROVED** [review]
Nice I think this helps a lot of people that do not have the latest CMake.

**Pull request merged**